webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala

```diff
@@ -2007,49 +2034,63 @@ class StringFormatter private (
       case None    => None
     }
 
+  def validateAndEscapeOptionalProjectShortname(
+    maybeString: Option[String]
+  ): Validation[ValidationException, Option[String]] =
+    maybeString match {
+      case None        => Validation.succeed(None)
+      case Some(value) => validateAndEscapeProjectShortname(value).map(Some(_))
```

Comment:
Again, consider removing those two methods entirely.

If you want to retain them, then the `validateAndEscapeOptionalProjectShortname(maybeString: Option[String],  errorFun: => Nothing)` should point to the `validateAndEscapeOptionalProjectShortname(maybeString: Option[String])` otherwise the logic which assumes the validity of a `None` input is duplicated.

I suppose same applies to the similar methods above, please check.

Reply:
All "optional" methods are gone now 🔥